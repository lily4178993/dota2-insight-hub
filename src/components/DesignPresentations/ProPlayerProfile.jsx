import React from 'react';
import PropTypes from 'prop-types';
import './proPlayerProfile.css';

const ProPlayerProfile = ({ data }) => (
  <div className="pro-player-profile">
    <h2>Dota 2 Pro Player Profile</h2>
    <div className="player-details">
      <div className="avatar">
        <img src={data.avatarfull} alt="Player Avatar" />
      </div>
      <div className="info">
        <h3>{data.personaname}</h3>
        <p>
          Steam ID:
          {' '}
          {data.steamid}
        </p>
        <p>
          Country:
          {' '}
          {data.loccountrycode}
        </p>
      </div>
    </div>
    <div className="team-details">
      <h3>Team</h3>
      <p>
        Team Name:
        {' '}
        {data.team_name}
      </p>
      <p>
        Team Tag:
        {' '}
        {data.team_tag}
      </p>
    </div>
    <div className="player-status">
      <h3>Player Status</h3>
      <p>
        Pro Player:
        {' '}
        {data.is_pro ? 'Yes' : 'No'}
      </p>
      <p>
        Locked:
        {' '}
        {data.is_locked ? 'Yes' : 'No'}
      </p>
      {data.is_locked && (
        <p>
          Locked Until:
          {' '}
          {data.locked_until}
        </p>
      )}
    </div>
  </div>
);

ProPlayerProfile.propTypes = {
  data: PropTypes.shape({
    avatarfull: PropTypes.string.isRequired,
    personaname: PropTypes.string.isRequired,
    steamid: PropTypes.string.isRequired,
    loccountrycode: PropTypes.string.isRequired,
    team_name: PropTypes.string.isRequired,
    team_tag: PropTypes.string.isRequired,
    is_pro: PropTypes.bool.isRequired,
    is_locked: PropTypes.bool.isRequired,
    locked_until: PropTypes.number.isRequired,
  }).isRequired,
};

export default ProPlayerProfile;
