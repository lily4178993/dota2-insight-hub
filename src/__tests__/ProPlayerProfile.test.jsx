import React from 'react';
import { render } from '@testing-library/react';
import ProPlayerProfile from '../components/DesignPresentations/ProPlayerProfile';

test('ProPlayerProfile component matches snapshot', () => {
  const testData = {
    account_id: 123,
    avatarfull: 'avatar-url',
    personaname: 'PlayerName',
    steamid: 'steam-id',
    loccountrycode: 'US',
    team_name: 'TeamName',
    team_tag: 'TT',
    is_pro: true,
    is_locked: false,
  };

  const { asFragment } = render(<ProPlayerProfile data={testData} />);

  expect(asFragment()).toMatchSnapshot();
});
