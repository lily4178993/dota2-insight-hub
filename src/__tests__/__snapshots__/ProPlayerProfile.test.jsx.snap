// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProPlayerProfile component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="player-profile-container"
  >
    <h2>
      Player Card
    </h2>
    <div
      class="player-profile"
    >
      <div
        class="player-details"
      >
        <div
          class="avatar"
        >
          <img
            alt="Player Avatar"
            src="avatar-url"
          />
        </div>
        <div
          class="info"
        >
          <h3>
            Pseudo: PlayerName
          </h3>
          <p>
            <strong>
              ID:
            </strong>
             123
          </p>
          <p>
            <strong>
              Steam ID:
            </strong>
             steam-id
          </p>
          <p>
            <strong>
              Country:
            </strong>
             US
          </p>
        </div>
      </div>
      <div
        class="player-other-details"
      >
        <div
          class="team-details"
        >
          <h3>
            Team
          </h3>
          <p>
            <strong>
              Name:
            </strong>
             TeamName
          </p>
          <p>
            <strong>
              Tag:
            </strong>
             TT
          </p>
        </div>
        <div
          class="player-status"
        >
          <h3>
            Player Status
          </h3>
          <p>
            <strong>
              Pro Player:
            </strong>
             Yes
          </p>
          <p>
            <strong>
              Locked:
            </strong>
             No
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
